birdhouse/components/weaver/config/magpie/weaver_hooks.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+These hooks will be running within Twitcher, using MagpieAdapter context, applied for Weaver requests.
+
+The code below can make use of any package that is installed by Magpie/Twitcher.
+"""
+
+import json
+from typing import TYPE_CHECKING
+
+from magpie.api.management.resource import resource_utils as ru
+from magpie.api.requests import get_user
+from magpie.constants import get_constant
+from magpie.permissions import Access, Permission
+from magpie.utils import get_header
+
+if TYPE_CHECKING:
+    from pyramid.request import Request
+    from pyramid.response import Response
+
+    from magpie.adapter import HookContext
+
+
+def is_admin(request):
+    # type: (Request) -> bool
+    admin_group = get_constant("MAGPIE_ADMIN_GROUP", settings_container=request)
+    if not request.user:  # no user authenticated (public)
+        return False
+    return admin_group in [group.group_name for group in request.user.groups]
+
+
+def add_x_wps_output_context(request):
+    # type: (Request) -> Request
+    """
+    Apply the ``X-WPS-Output-Context`` for saving outputs in the user-context WPS-outputs directory.
+    """
+    header = get_header("X-WPS-Output-Context", request.headers)
+    # if explicitly provided, ensure it is permitted (admin allow any, otherwise self-user reference only)
+    if header is not None:
+        if request.user is None:
+            header = "public"
+        else:
+            if not is_admin(request):
+                # override disallowed writing to other location
+                # otherwise, up to admin to have writen something sensible
+                header = "user-" + str(request.user.id)
+    else:
+        if request.user is None:
+            header = "public"
+        else:
+            header = "user-" + str(request.user.id)
+    request.headers["X-WPS-Output-Context"] = header
+    return request
+
+
+def filter_allowed_processes(response, context):
+    # type: (Response, HookContext) -> Response
+    """
+    Filter processes returned by Weaver response according to allowed resources by user.
+    """
+    if "application/json" in response.content_type:
+        body = response.json
+        if "processes" in body:
```

Comment:
Can add as comment a sample json body?  So it's easier to follow the code below that extract processes id or verbose string, and seems to do something with the children.

Reply:
To follow up on @dbyrns comment about supported `processes` structure, showing a sample json body here will also clarify what is the expected `processes` structure.

Reply:
Added in 28ad95c